clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4143,19 +4140,27 @@ public RemoveMembersFromConsumerGroupResult removeMembersFromConsumerGroup(Strin
         String reason = options.reason() == null || options.reason().isEmpty() ?
             DEFAULT_LEAVE_GROUP_REASON : JoinGroupRequest.maybeTruncateReason(options.reason());
 
-        List<MemberIdentity> members;
-        if (options.removeAll()) {
-            members = getMembersFromGroup(groupId, reason);
-        } else {
-            members = options.members().stream()
-                .map(m -> m.toMemberIdentity().setReason(reason))
-                .collect(Collectors.toList());
+        final SimpleAdminApiFuture<CoordinatorKey, Map<MemberIdentity, Errors>> future =
+                RemoveMembersFromConsumerGroupHandler.newFuture(groupId);
+
+        KafkaFutureImpl<List<MemberIdentity>> f;
+        if (options.removeAll()) f = getMembersFromGroup(groupId, reason);
```

Comment:
nit: I would recommend not omitting the curly braces, or at the very least, it should break the line.